perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Metacentrum.java

```diff
@@ -142,17 +143,10 @@ public void canBeApproved(PerunSession session, Application app) throws PerunExc
 				// get eligible date + 1 year
 				Date eligibleDate = df.parse(eligibleString);
 
-				Calendar c = Calendar.getInstance();
-				c.setTime(eligibleDate);
-				c.add(Calendar.YEAR, 1);
-				Date eligibleDatePlusYear = c.getTime();
-
-				// get now
-				Calendar cal = Calendar.getInstance();
-				Date now = cal.getTime();
+				LocalDateTime timeInOneYear = LocalDateTime.ofInstant(eligibleDate.toInstant(), ZoneId.systemDefault()).plusYears(1);
```

Comment:
Can you please verify, that this logic doesn't add the value in hours ? Since sourcing value is a date/time without the zone, but was generated in our zone (+1/+2 in summer). So if you again apply zone specific offset to the value, then it can add an hour or two, which doesn't correspond with the logic, that we hold user eligible 1 year after the timestamp.

Reply:
Yes, from what I tried, it does not change hours during the conversions. I think that since the SimpleDateFormat uses the same default time zone to create Date as LocalDateTime does, there should not be any problems.

Reply:
Ok